cmd/run.go

```diff
@@ -197,6 +196,16 @@ a commandline interface for interacting with it.`,
 				return err
 			}
 
+			registry := stats.NewRegistry(engine.Samples)
+			builtInMetrics := metrics.RegisterBuiltinMetrics(registry)
+			globalCtx = stats.WithRegistry(globalCtx, registry)
+			globalCtx = metrics.WithBuiltinMetrics(globalCtx, builtInMetrics)
+
+			lingerCtx, lingerCancel := context.WithCancel(globalCtx)
+			defer lingerCancel()
+			runCtx, runCancel := context.WithCancel(lingerCtx)
+			defer runCancel()
```

Comment:
This hacks were needed as I need to have the builtinMetrics in both `setup`, `teardown`, `handleSummary`  and all the outputs, so it needs to be as early as possible, but I also need the engine's Samples as I didn't want to move that as well

Reply:
for starters ... lib.State is not around in the init context

Reply:
Fair enough, but you have this: https://github.com/k6io/k6/blob/7da10ac81128940e0828d9aca2e955f4e046c742/js/common/initenv.go#L31-L33

It's actually even better, since that way you can ensure that metrics can be registered only in the init env :man_shrugging: